src/System.Net.Requests/src/System/Net/HttpWebRequest.cs

```diff
@@ -73,6 +73,10 @@ public class HttpWebRequest : WebRequest, ISerializable
         private bool _preAuthenticate;
         private DecompressionMethods _automaticDecompression = HttpHandlerDefaults.DefaultAutomaticDecompression;
 
+        private static readonly object s_syncRoot = new object();
+        private static HttpClient s_cachedHttpClient;
```

Comment:
Yes, I missed volatile keyword for HttpClient which is required here since it's read outside the lock. But, could you please clarify why volatile is required for s_syncRoot? AFAIK, constructors have full-fence semantics, so all threads should see the fully initialized object under any circumstances. Additionally, our [official Singleton guideline](https://docs.microsoft.com/en-us/previous-versions/msp-n-p/ff650316(v=pandp.10)?redirectedfrom=MSDN) doesn't have volatile for sync root. Or, is there any issue with some specific hardware architecture?

Reply:
I was commenting on s_cachedHttpClient.  s_syncRoot does not / should not be volatile.

Reply:
Done.